dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -1077,18 +1079,33 @@ def apply(self, sdfg):
                             and node.desc(nested_sdfg).transient and
                             not isinstance(node.desc(nested_sdfg), data.View)):
                         arrname = node.data
-                        if arrname not in transients and not scope_dict[node]:
-                            arrobj = nested_sdfg.arrays[arrname]
-                            nested_sdfg.arrays['__' + arrname + '_out'] = arrobj
-                            outer_sdfg.arrays[arrname] = dc(arrobj)
-                            transients[arrname] = '__' + arrname + '_out'
-                        node.data = '__' + arrname + '_out'
-        for arrname in inputs.keys():
-            nested_sdfg.arrays.pop(arrname)
-        for arrname in outputs.keys():
-            nested_sdfg.arrays.pop(arrname, None)
-        for oldarrname, newarrname in transients.items():
-            nested_sdfg.arrays.pop(oldarrname)
+                        if not scope_dict[node]:
+                            arrname_nested = f"__{arrname}_out"
+                            node.data = arrname_nested
+                            if arrname not in transients:
+                                arrobj = nested_sdfg.arrays[arrname]
+                                outer_sdfg.arrays[arrname] = dc(arrobj)
+                                nested_sdfg.arrays[arrname_nested] = arrobj
+                                transients[arrname] = arrname_nested
+
+        # Catch data containers that we didn't find on any access nodes, and add
+        # them as inputs. This can happen when a scalar input is used on an
+        # interstate edge, and thus doesn't appear in the dataflow.
+        nested_data = set(
```

Comment:
Since we are here...
In the npbench_fpga branch there is something similar: we keep track of input/output transients and we add them as input/output connectors (https://github.com/spcl/dace/blob/61748d925fa9218ebfab4787fced0d15f72e2d9a/dace/transformation/interstate/sdfg_nesting.py#L1168-L1184).

I think that with the current changes, it adds the transients only to the output transients, but not the input. If it makes sense, can you add that?

Reply:
Done in 3196301b